pkg/virt-operator/install-strategy/create.go

```diff
@@ -1384,6 +1386,66 @@ func addOrRemoveSSC(targetStrategy *InstallStrategy,
 	return nil
 }
 
+func syncPodDisruptionBudgetForDeployment(deployment *appsv1.Deployment, clientset kubecli.KubevirtClient, kv *v1.KubeVirt, expectations *util.Expectations, stores util.Stores) error {
+	podDisruptionBudget := components.NewPodDisruptionBudgetForDeployment(deployment)
+
+	imageTag := kv.Status.TargetKubeVirtVersion
+	imageRegistry := kv.Status.TargetKubeVirtRegistry
+
+	injectOperatorMetadata(kv, &podDisruptionBudget.ObjectMeta, imageTag, imageRegistry)
+
+	var cachedPodDisruptionBudget *policyv1beta1.PodDisruptionBudget
+
+	pdbClient := clientset.PolicyV1beta1().PodDisruptionBudgets(deployment.Namespace)
+
+	obj, exists, _ := stores.PodDisruptionBudgetCache.Get(podDisruptionBudget)
+
+	if exists {
```

Comment:
You're referring to [here](https://github.com/dhiller/kubevirt/blob/865b2420cbc6134cfa45ebf21f1e0b1a8c87acee/pkg/virt-operator/install-strategy/create.go#L221) and [here](https://github.com/dhiller/kubevirt/blob/865b2420cbc6134cfa45ebf21f1e0b1a8c87acee/pkg/virt-operator/install-strategy/create.go#L158), right?

I found this less confusing to read because the whole `if exists` block applies only if the item exists in cache. I found the other blocks more confusing to read because on a closer look the pattern is like

```
obj, exists := getfromcache
if exists {
  ... cast object to use it later in other blocks
}
if !exists {
  ... do things
} else if !objectMatchesVersion {
  ... do other things
} else {
  ... do even other things
}
```

This confused me insofar as I could not really see where the casted object was used at first. So I changed it into this. 

If you think this is more confusing, I'll change it to match to the other patterns. But I think the scope of usage for `cachedPodDisruptionBudget` is mroe clear now. I could even remove the declaration from line 1397 and convert 1404 into a `:=`. What do you think?

Reply:
You're referring to [here](https://github.com/dhiller/kubevirt/blob/865b2420cbc6134cfa45ebf21f1e0b1a8c87acee/pkg/virt-operator/install-strategy/create.go#L221) and [here](https://github.com/dhiller/kubevirt/blob/865b2420cbc6134cfa45ebf21f1e0b1a8c87acee/pkg/virt-operator/install-strategy/create.go#L158), right?

I found this less confusing to read because the whole `if exists` block applies only if the item exists in cache. I found the other blocks more confusing to read because on a closer look the pattern is like

```
obj, exists := getfromcache
if exists {
  ... cast object to use it later in other blocks
}
if !exists {
  ... do things
} else if !objectMatchesVersion {
  ... do other things
} else {
  ... do even other things
}
```

This confused me insofar as I could not really see where the casted object was used at first. So I changed it into this. 

If you think this is more confusing, I'll change it to match to the other patterns. But I think the scope of usage for `cachedPodDisruptionBudget` is mroe clear now. I could even remove the declaration from line 1397 and convert 1404 into a `:=`. What do you think?

Reply:
> You're referring to here and here, right?

Not only, the createOrUpdate* funcs use the same pattern as well.
Mh, not sure which pattern is easier to read, I have no preference on that. But IMHO we should use the same everywhere.

Reply:
OK, I'll change it back to the other pattern.

Reply:
Done.